apps/OboeTester/app/src/main/cpp/FullDuplexStream.cpp

```diff
@@ -112,10 +123,13 @@ oboe::Result FullDuplexStream::start() {
     // Determine maximum size that could possibly be called.
     int32_t bufferSize = getOutputStream()->getBufferCapacityInFrames()
             * getOutputStream()->getChannelCount();
-    if (bufferSize > mBufferSize) {
-        mInputBuffer = std::make_unique<float[]>(bufferSize);
-        mBufferSize = bufferSize;
-    }
+    mInputConverter = std::make_unique<FormatConverterBox>(bufferSize,
+    getInputStream()->getFormat(),
+    oboe::AudioFormat::Float);
+    mOutputConverter = std::make_unique<FormatConverterBox>(bufferSize,
+            oboe::AudioFormat::Float,
+            getOutputStream()->getFormat());
+
```

Comment:
I'm sure there's a really obvious answer here but why can't the conversion be done in a single stage, rather than having one converter for the input and one for the output with Float being the intermediate format?

Reply:
Because I wanted to do the full duplex processing in Float and the input or output streams may be a different format. So I need to process on each end.